src/renderer/components/dock/dock.store.ts

```diff
@@ -181,11 +181,15 @@ export class DockStore implements DockStorageState {
     if (this.height > this.maxHeight) this.height = this.maxHeight;
   }
 
-  onResize(callback: () => void, options?: IReactionOptions) {
+  onResize(callback: () => void, options?: Omit<IReactionOptions<void, boolean>, "equals">) {
+    (options as IReactionOptions<void, boolean>).equals = undefined;
+
     return reaction(() => [this.height, this.fullSize], callback, options);
   }
 
-  onTabChange(callback: (tabId: TabId) => void, options?: IReactionOptions) {
+  onTabChange(callback: (tabId: TabId) => void, options?: Omit<IReactionOptions<void, boolean>, "equals">) {
+    (options as IReactionOptions<void, boolean>).equals = undefined;
+
```

Comment:
If `options` is optional why you try to assign `options.equals` to `undefined` and why it's needed at all?

```suggestion
  onTabChange(callback: (tabId: TabId) => void, options?: CustomReactionOpts) {
```

Reply:
To block the changing of the `equals` field. I guess that isn't really needed.